packages/core/src/routes/session.ts

```diff
@@ -105,7 +105,24 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
         return next();
       }
 
-      await signInWithSocial(ctx, provider, { connectorId, code, result });
+      await signInWithSocial(ctx, provider, { connectorId, code });
+
+      return next();
+    }
+  );
+
+  router.post(
+    '/session/sign-in/social-related-user',
```

Comment:
will it be better to use `bind-...` to align with product design?

Reply:
LOG-1597